spec/unit/provider/package/dnfmodule_spec.rb

```diff
@@ -202,12 +202,45 @@
     end
   end
 
+  describe "when disabling a module" do
```

Comment:
I think you can use the `describe` block above and rename it from `when installing a module` to `when ensuring a module`, so we don't duplicate the code too much.

You can change this block into a `context` block like this one then https://github.com/puppetlabs/puppet/pull/8137/files#diff-8564081f1ab7466e376357ec7d6ad71aR182